src/Microsoft.FeatureManagement/Targeting/TargetingEvaluator.cs

```diff
@@ -0,0 +1,370 @@
+            {
+                paramName = $"{targetingSettings.Audience}.{targetingSettings.Audience.DefaultRolloutPercentage}";
+
+                reason = OutOfRange;
+
+                return false;
+            }
+
+            if (targetingSettings.Audience.Groups != null)
+            {
+                int index = 0;
+
+                foreach (GroupRollout groupRollout in targetingSettings.Audience.Groups)
+                {
+                    index++;
+
+                    if (groupRollout.RolloutPercentage < 0 || groupRollout.RolloutPercentage > 100)
+                    {
+                        //
+                        // Audience.Groups[1].RolloutPercentage
+                        paramName = $"{targetingSettings.Audience}.{targetingSettings.Audience.Groups}[{index}].{groupRollout.RolloutPercentage}";
+
+                        reason = OutOfRange;
+
+                        return false;
+                    }
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Determines if a given context id should be targeted based off the provided percentage range
+        /// </summary>
+        public static bool IsTargeted(ITargetingContext targetingContext, double from, double to, string seed, bool ignoreCase, string hint)
+        {
+            byte[] hash;
+
+            string userId = ignoreCase ?
+                targetingContext.UserId.ToLower() :
+                targetingContext.UserId;
+
+            string contextId;
+
+            if (!string.IsNullOrEmpty(seed))
+            {
+                contextId = $"{userId}\n{seed}";
+            }
+            else
+            {
+                contextId = $"{userId}\n{hint}";
+            }
+
+            using (HashAlgorithm hashAlgorithm = SHA256.Create())
+            {
+                hash = hashAlgorithm.ComputeHash(Encoding.UTF8.GetBytes(contextId).ToArray());
+            }
+
+            //
+            // Use first 4 bytes for percentage calculation
+            // Cryptographic hashing algorithms ensure adequate entropy across hash
+            uint contextMarker = BitConverter.ToUInt32(hash, 0);
+
+            double contextPercentage = (contextMarker / (double)uint.MaxValue) * 100;
+
+            return contextPercentage >= from && contextPercentage <= to;
+        }
+
+        /// <summary>
+        /// Determines if a given context id should be targeted based off the provided percentage
+        /// </summary>
+        /// <param name="contextId">A context identifier that determines what the percentage is applicable for</param>
+        /// <param name="percentage">The total percentage of possible context identifiers that should be targeted</param>
+        /// <returns>A boolean representing if the context identifier should be targeted</returns>
+        private static bool IsTargeted(string contextId, double percentage)
```

Comment:
Could have this call the (from, to) with (0, percentage) or have them both use the same helper. Would have to rewrite the other function slightly. 

Feel free to keep them separate if you prefer, I just liked the idea of demonstrating how percentage is just percentile with no seed.